rasterio/fill.py

```diff
@@ -49,6 +50,16 @@ def fillnodata(
     out : numpy ndarray
         The filled raster array.
     """
+    if mask is None and hasattr(image, 'mask'):  # pragma: no cover
```

Comment:
@sgillies these should have corresponding tests, and be covered.  I think we should reserve `#pragma: no cover` for exception cases that we unable to test directly.

Reply:
Looks like your new test case does cover this?  Why exclude it from coverage, then?

Reply:
I'm not trying to game the coverage, I swear :)

I'm trying to dodge the branch warning in these cases where I have no `else` clause.

Reply:
Would `# pragma: no branch` do the trick here?

http://coverage.readthedocs.io/en/latest/branch.html#structurally-partial-branches